test/test_vmap.py

```diff
@@ -3581,7 +3566,7 @@ def test_feature_dropout(self, device, randomness):
         ]
 
         for op in supported_ops:
-            passed = torch.ones([4, 3, 3, 3, 3])
+            passed = torch.ones([4, 3, 3, 3, 3], device=device)
```

Comment:
We should really test batch_dim != 0 as well as batch_dim == 0 (IIRC all the existing tests test batch_dim == 0)?

Reply:
While I'm on the "things to improve testing train": we should have a one-off manual test with just a single op that tests multiple vmaps and different randomness behavior to check that the mode mechanism works with multiple levels.

That is, we should check `vmap(vmap(f, randomness=different), randomness=same)` in all 9 combinations of randomness = {same, different, error} x {same, different, error}.